lib/matplotlib/cbook.py

```diff
@@ -1012,18 +1012,26 @@ def mkdirs(newdir, mode=0o777):
         > mkdir -p NEWDIR
         > chmod MODE NEWDIR
     """
-    try:
-        if not os.path.exists(newdir):
-            parts = os.path.split(newdir)
-            for i in range(1, len(parts) + 1):
-                thispart = os.path.join(*parts[:i])
-                if not os.path.exists(thispart):
-                    os.makedirs(thispart, mode)
-
-    except OSError as err:
-        # Reraise the error unless it's about an already existing directory
-        if err.errno != errno.EEXIST or not os.path.isdir(newdir):
-            raise
+    # this functionality is now in core python as of 3.2
+    if six.PY3:
+        os.makedirs(newdir, mode=mode, exist_ok=True)
+        return
```

Comment:
(nit): I know it's equivalent, but I think I'd prefer to see the Python 2 stuff in an `else` clause so it's more obvious it only runs on Python 2.